bouncer/src/repo/manipulator/modelconvertor/import/odaHelper/data_processor_rvt.cpp

```diff
@@ -345,518 +370,760 @@ void DataProcessorRvt::initLabelUtils() {
 	}
 }
 
-std::string DataProcessorRvt::unitsToString(const OdBm::DisplayUnitType::Enum &units) {
-	switch (units) {
-	case  OdBm::DisplayUnitType::Enum::DUT_UNDEFINED:
-	case  OdBm::DisplayUnitType::Enum::DUT_CUSTOM:
-	case  OdBm::DisplayUnitType::Enum::DUT_GENERAL:
-	case  OdBm::DisplayUnitType::Enum::DUT_FIXED:
-	case OdBm::DisplayUnitType::Enum::DUT_CURRENCY:
+boost::optional<std::string> DataProcessorRvt::unitsToString(const OdBmForgeTypeId& units) {
+	if (units.isEmpty()) { // Use if not if..else in this function because otherwise we reach compiler limits on nested blocks
```

Comment:
I'm assuming they don't provide a units to units label of some sort still? 😢 

I tried looking for one before I coded this, and failed

Reply:
> I tried looking for one before I coded this, and failed

I thought that had come from a sample or at least there was a reason we were overriding, not that it forced that to be written by hand! 😧 

> I'm assuming they don't provide a units to units label

There is now a way to do this using `getLabelForSymbol`, though its very obscure...

Passing the unit spec objects directly from the parameter descriptor fails silently, because while they describe the unit, they don't actually describe a _symbol_ itself (which is what that method is expecting) (nor can they return the symbol, which would be useful...)
It turns out the format options object for the unit type/spec object needs to be retrieved, *that* object will return another spec type object, but this one for the symbol itself, and *this* object can be passed to `getLabelForSymbol` to return a valid string.

This mechanism is what it is used in the `OdBmSampleLabelUtilsPE::format` method, which fortunately we have the source to, so I've implemented something similar for `processParameter`.